charts/azuremonitor-containers/templates/omsagent-priorityclass.yaml

```diff
@@ -0,0 +1,22 @@
+{{- if and (ne .Values.omsagent.secret.key "<your_workspace_key>") (ne .Values.omsagent.secret.wsid "<your_workspace_id>") (or (ne .Values.omsagent.env.clusterName "<your_cluster_name>") (ne .Values.omsagent.env.clusterId "<your_cluster_id>") (ne .Values.Azure.Cluster.ResourceId "<your_cluster_id>") )}}
+# This pod priority class is used for daemonsets to allow them to have priority
+# over pods that can be scheduled elsewhere.  Without a priority class, it is
+# possible for a node to fill up with pods before the daemonset pods get to be
+# created for the pod or get scheduled.  Note that pods are not "daemonset"
+# pods - they are just pods created by the daemonset controller but they have
+# a specific affinity set during creation to the specific node each pod was
+# created to run on (daemonset controller takes care of that)
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: omsagent
+  # Priority classes don't have labels :-)
+  annotations:
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    component: oms-agent
+value: {{ .Values.omsagent.priority }}
```

Comment:
check needed for ensuring .Values.omsagent.priority is a number ?

Reply:
Helm will do those checks as part of the validation against the schema when applying to a cluster.